sockeye/training.py

```diff
@@ -599,6 +716,12 @@ def _evaluate(self, training_state, val_iter, val_metric):
         val_iter.reset()
         val_metric.reset()
 
+        state_names = ['training_state.updates']
```

Comment:
This could use a comment why we do this. It could also use the _set_state() function you define above.
Also why are we resetting the update count back to 0 here?